expo/scheduler/views.py

```diff
@@ -593,117 +598,143 @@ def contact_info(request,
-                  'Rehearsal(s)']
-        contact_info = []
-        for c in contacts:
-            act = c.as_subtype
-            performer = act.performer
-            contact_info.append(
-                [act.title,
-                 str(performer),
-                 str(performer.contact),
-                 act.contact_email,
-                 act.performer.contact.phone,
-                 act.accepted,
-                 ",".join(map(str, act.get_scheduled_shows())),
-                 ",".join(map(str, act.get_scheduled_rehearsals()))]
-            )
-    elif participant_type == 'Volunteers':
-        contacts = Worker.objects.filter(role='Volunteer')
-        contacts = [w for w in contacts if w.allocations.count() > 0]
-        header = ['Name',
-                  'Phone',
-                  'Email',
-                  'Volunteer Category',
-                  'Volunteer Role',
-                  'Event']
-        from gbe_forms_text import volunteer_interests_options
-        volunteer_categories = dict(volunteer_interests_options)
-        contact_info = []
-        for c in contacts:
-            profile = c.item.profile
-            event = c.allocations.first().event
-            try:
-                parent_event = event.container_event.parent_event
-            except:
-                parent_event = event
+def contact_performers(conference=None):
+    if not conference:
+        conference = get_current_conference()
+    from gbe.models import Act
+    contacts = [act.actitem_ptr for act in Act.objects.filter(
+        conference=conference)]
+    header = ['Act',
+              'Performer',
+              'Profile',
+              'email',
+              'Phone',
+              'status',
+              'Show(s)',
+              'Rehearsal(s)']
+    contact_info = []
+    for c in contacts:
+        act = c.as_subtype
+        performer = act.performer
+        contact_info.append(
+            [act.title,
+             str(performer),
+             str(performer.contact),
+             act.contact_email,
+             act.performer.contact.phone,
+             act.accepted,
+             ",".join(map(str, act.get_scheduled_shows())),
+             ",".join(map(str, act.get_scheduled_rehearsals()))]
+        )
+    return header, contact_info
+
+
+def contact_volunteers(conference=None):
+    header = ['Name',
+              'Phone',
+              'Email',
+              'Volunteer Category',
+              'Volunteer Role',
+              'Event']
+    volunteer_categories = dict(volunteer_interests_options)
+    if not conference:
+        conference = get_current_conference()
+    from gbe.models import Volunteer
+    contacts = filter(lambda worker: worker.allocations.count() > 0,
+                      [vol.profile.workeritem_ptr.worker_set.first() for vol in
+                       Volunteer.objects.filter(conference=conference)
+                       if vol.profile.workeritem_ptr.worker_set.exists()])
```

Comment:
Let me check the logic on this... is this looking for ?
- volunteers who filled out the bid...
- and who are booked into an event as volunteers...

Right?

Does this case ALSO cover - people who were booked for a volunteer event, but who didn't fill out a form?  I think that's possible in the code and should be represented here...